src/osd/PG.cc

```diff
@@ -3808,14 +3808,8 @@ void PG::replica_scrub(
     return;
   }
 
-  // compensate for hobject_t's with wrong pool from sloppy hammer OSDs
-  hobject_t start = msg->start;
-  hobject_t end = msg->end;
-  start.pool = info.pgid.pool();
-  end.pool = info.pgid.pool();
-
   build_scrub_map_chunk(
-    map, start, end, msg->deep, msg->seed,
+    map, msg->start, msg->end, msg->deep, msg->seed,
```

Comment:
With my change the start already has the pool number in it being passed from caller.  The only affect my code has is that as you noted above I didn't set "scrubber.end."  I couldn't see a reason for it, so I left it out.  As a matter of fact I always saw end shown as MAX in the log even when I set the pool there too.